File: dbpu-l9/mnozenie.java
import java.util.Random;

public class mnozenie {
    Integer[][] A, B, C;
    private int size;
	private Random rand = new Random();
    
    public mnozenie(int defaultSize) {
    	size = defaultSize;
    	A = new Integer[defaultSize][defaultSize];
    	B = new Integer[defaultSize][defaultSize];
    	C = new Integer[defaultSize][defaultSize];

        for (int i = 0; i < defaultSize; i++) {
            for (int j = 0; j < defaultSize; j++) {
            	A[i][j] = rand.nextInt(51);
            	B[i][j] = rand.nextInt(51);
                C[i][j] = 0;
            }
        }
    	
    }
    public Integer[][] matricesMultiply() {
        for (int i = 0; i < this.size; i++) { 
            for (int j = 0; j < this.size; j++) {
                for (int k = 0; k < this.size; k++) {
                    this.C[i][j] += this.A[i][k] * this.B[k][j];
                }
            }
        }

        return C;
    }

    public static void main(String[] args) {
    	int defaultSize = 500;
        mnozenie matrix = new mnozenie(defaultSize);
        
    	long startTime = System.currentTimeMillis();
        Integer[][] result = matrix.matricesMultiply();

/*        for (int i = 0; i < defaultSize; i++) {
            for (int j = 0; j < defaultSize; j++)
                System.out.print(result[i][j] + " ");
            
            System.out.println();
        }*/

        long endTime = System.currentTimeMillis();
        long executionTime = endTime - startTime;
        System.out.println("Execution time: " + executionTime/1000 + " seconds");
    }
}

File: dbpu-l9/mnozenie.js
var defaultSize = parseInt(process.argv[2]) || 500;
var matrixA = new Array(defaultSize),
	matrixB = new Array(defaultSize),
	matrixC = new Array(defaultSize);

for(var i = 0; i < defaultSize; i++) {
	matrixA[i] = new Array();
	matrixB[i] = new Array();
	matrixC[i] = new Array();
}

function getRandomNumber(min, max){
    return Math.floor(Math.random() * (max - min + 1)) + min;
}

function fillMatrices () {
	for(var i = 0; i < defaultSize; i++) {
		for(var j = 0; j < defaultSize; j++) {
			matrixA[i][j] = getRandomNumber(0, 50); 
			matrixB[i][j] = getRandomNumber(0, 50);
			matrixC[i][j] = 0;
		}
	}
}

function multiplyMatrices () {
	for(var i = 0; i < defaultSize; i++) {
		for(var j = 0; j < defaultSize; j++) {
			for(var k = 0; k < defaultSize; k++) {
				matrixC[i][j] += matrixA[i][k] * matrixB[k][j];
			}
		}
	}
}

var startTime = new Date().getTime();

fillMatrices();
multiplyMatrices();

var endTime = new Date().getTime();
var executionTime = endTime - startTime;
console.log('Execution time: ', executionTime/1000, 'seconds');

// To print results
//console.table ? console.table(matrixC) : console.log(matrixC);
